internal/storage/fs/snapshot.go

```diff
@@ -626,19 +703,56 @@ func (ss *storeSnapshot) GetEvaluationDistributions(ctx context.Context, ruleID
 }
 
 func (ss *storeSnapshot) GetEvaluationRollouts(ctx context.Context, namespaceKey, flagKey string) ([]*storage.EvaluationRollout, error) {
-	panic("not implemented")
+	ns, ok := ss.ns[namespaceKey]
+	if !ok {
+		return nil, errs.ErrNotFoundf("namespaced %q", namespaceKey)
+	}
+
+	rollouts, ok := ns.evalRollouts[flagKey]
+	if !ok {
+		return nil, errs.ErrNotFoundf(`flag "%s/%s"`, namespaceKey, flagKey)
+	}
+
+	return rollouts, nil
 }
 
 func (ss *storeSnapshot) GetRollout(ctx context.Context, namespaceKey, id string) (*flipt.Rollout, error) {
-	panic("not implemented")
+	ns, err := ss.getNamespace(namespaceKey)
+	if err != nil {
+		return nil, err
+	}
+
+	rollout, ok := ns.rollouts[id]
+	if !ok {
+		return nil, errs.ErrNotFoundf(`rollout "%s/%s"`, namespaceKey, id)
+	}
+
+	return rollout, nil
 }
 
-func (ss *storeSnapshot) ListRollouts(ctx context.Context, namespaceKey, flagKey string, opts ...storage.QueryOption) (storage.ResultSet[*flipt.Rollout], error) {
-	panic("not implemented")
+func (ss *storeSnapshot) ListRollouts(ctx context.Context, namespaceKey, flagKey string, opts ...storage.QueryOption) (set storage.ResultSet[*flipt.Rollout], err error) {
+	ns, err := ss.getNamespace(namespaceKey)
+	if err != nil {
+		return set, err
+	}
+
+	rollouts := make([]*flipt.Rollout, 0, len(ns.rollouts))
+	for _, rollout := range ns.rollouts {
+		rollouts = append(rollouts, rollout)
```

Comment:
These need filtering by flag key (+ a test ideally with rollouts for more than one flag key)

Reply:
Could either filter here or index the rollouts by flag when you build the snapshot.

Reply:
@GeorgeMac Ah good catch, thank you!